### PR TITLE
Add script to deploy OptimismMintableERC20 USDT token using OptimismMintableERC20Factory

### LINKED ISSUE #153: Add script to deploy OptimismMintableERC20 USDT token using OptimismMintableERC20Factory
### Description

Using a predeployed factory contract to deploy standard OP tokens is convenient and secure. Having a script to do that diminishes the effort even further.

### Acceptance Criteria

Add a script with common environment variables to deploy USDT on Lisk testnet and mainnet

### Additional Information
Predeployed OptimismMintableERC20Factory contract addresses on Lisk testnet and mainnet can also be found [here](https://docs.lisk.com/contracts)

### PR DESCRIPTION
**Pull request by Incede**
### What was the problem?

This PR resolves #153

### How was it solved?

Added script with environment variable for testnet and mainnet

### How was it tested?

 - Run the script from inside scripts folder `bash 6_deployUSDTContract.sh `
 - Add `--nonce` flag to specify the correct nonce in the command in script, in case there is a nonce mismatch reported
 - Supply private key at the prompt to sign and send the transaction
 - Wait for the process to finish, check the output transaction hex in block explorer

### CODE REVIEW

File: .env.factory.mainnet [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# L2 RPC URL, e.g. Infura, Alchemy, or your own node
+L2_RPC_URL=https://lisk.drpc.org
+
+# Name of the L2 token
+TOKEN_NAME=Tether USD
+
+#Symbol of the L2 token
+TOKEN_SYMBOL=USDT
+
+#Decimals in the L2 token
```
Comment (shuse2) [inline .env.factory.mainnet:13]: can be removed and add to the script?

File: .env.factory.mainnet [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# L1 token contract address to be bridged to L2, defaults to USDT
```
Comment (shuse2) [inline .env.factory.mainnet:1]: we shouldn't make new file for this, but we should use .env.testnet and .env.mainnet?

File: .env.factory.mainnet [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+REMOTE_TOKEN_ADDR=0xdAC17F958D2ee523a2206206994597C13D831ec7
```
Comment (shuse2) [inline .env.factory.mainnet:2]:
```suggestion
REMOTE_TOKEN_ADDR_USDT=0xdAC17F958D2ee523a2206206994597C13D831ec7
```

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: script/6_deployUSDTContract.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+source ../.env.testnet
```
Comment (shuse2) [inline script/6_deployUSDTContract.sh:3]:
```suggestion
echo "Setting environment variables..."
source .env
echo "Done."
```

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: script/6_deployUSDTContract.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+echo "Setting environment variables..."
+source .env
```
Comment (ricott1) [inline script/6_deployUSDTContract.sh:4]: It looks like you want to run this script from the project root. Other deployment scripts are rather meant to be run from the script folder (and have an extra `cd ../` command at the beginning). I think we should be consistent

File: script/6_deployUSDTContract.sh [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+echo "Done."
+
+echo "Deploying OptimismMintableERC20 USDT token smart contract..."
+# Use flag --nonce to specify suitable nonce in case of nonce mismatch
+cast send --rpc-url $L2_RPC_URL "0xc0D3c0d3C0d3c0d3c0D3c0d3c0D3c0D3c0D30012" "createOptimismMintableERC20WithDecimals(address,string,string,uint8)" $REMOTE_TOKEN_ADDR_USDT "Tether USD" "USDT" 6 -i
```
Comment (ricott1) [inline script/6_deployUSDTContract.sh:9]: Isn't `0xc0D3c0d3C0d3c0d3c0D3c0d3c0D3c0D3c0D30012` the implementation? Should be run the script to the proxy?

**Review by matjazv — APPROVED** [review]
Except @ricott1 comment about `extra cd ../` it looks fine!

The only question I have is if we want to automatically save newly deployed contract address somewhere like we do it for other contracts? (eg. `deployment/testnet/l2addresses.json`)

**Comment by Incede** [comment]
> Except @ricott1 comment about `extra cd ../` it looks fine!
> 
> The only question I have is if we want to automatically save newly deployed contract address somewhere like we do it for other contracts? (eg. `deployment/testnet/l2addresses.json`)

The thing is most of the times (atleast on Lisk Sepolia) `cast` doesn't return the deployed contract address but only a transaction hex which may have been added to the mempool

File: script/6_deployUSDTContract.sh (hunk 6 above)
Comment (ricott1) [reply script/6_deployUSDTContract.sh:9]: I just tested it and you are right

File: script/6_deployUSDTContract.sh (hunk 5 above)
Comment (Incede) [reply script/6_deployUSDTContract.sh:4]: Would instead `source ../.env` make sense ?
if we do `cd../` to go to root then would need to come back to script dir again.

Comment (ricott1) [reply script/6_deployUSDTContract.sh]: I have no preference, but I would be consistent across all scripts

Comment (Incede) [reply script/6_deployUSDTContract.sh:8]: I updated per your suggestion

Comment (shuse2) [reply script/6_deployUSDTContract.sh:8]: maybe then we should also add
```
echo "Instructing the shell to exit immediately if any command returns a non-zero exit status..."
set -e
echo "Done."
```

Comment (shuse2) [reply script/6_deployUSDTContract.sh:8]: just for consistency. it's one command so it doesn't change anything

**Review by shuse2 — APPROVED** [review]

**Review by ricott1 — APPROVED** [review]

**Pull request merged**